crates/revm/src/optimism/processor.rs

```diff
@@ -84,7 +79,7 @@ impl<'a> BlockExecutor for EVMProcessor<'a> {
 
         let mut cumulative_gas_used = 0;
         let mut receipts = Vec::with_capacity(block.body.len());
-        for (transaction, sender) in block.body.iter().zip(senders) {
+        for (transaction, sender) in block.body.iter().zip(block.senders.iter()) {
```

Comment:
there's an iterator function for this `BlockWithSenders::transactions_with_sender`